app/src/main/java/org/p2p/wallet/home/ui/select/SelectTokenFragment.kt

```diff
@@ -72,14 +72,16 @@ class SelectTokenFragment :
         }
     }
 
-    override fun showTokens(items: List<Token>, filtered: Boolean) {
+    override fun showTokens(items: List<Token>) {
         tokenAdapter.setItems(items)
-        with(binding) {
-            val isEmpty = items.isEmpty()
-            tokenRecyclerView.isVisible = !isEmpty
-            emptyTextView.isVisible = isEmpty && !filtered
-            messageTextView.isVisible = isEmpty && filtered
-        }
+    }
+
+    override fun setEmptyViewVisibility(visible: Boolean) {
+        binding.emptyTextView.isVisible = visible
+    }
+
+    override fun setNoSearchedTokensViewVisibility(visible: Boolean) {
+        binding.messageTextView.isVisible = visible
```

Comment:
well, it's not message, it's more like `tokenNotFoundTextView`, not an ordinary message